### PR TITLE
Search List Envelopes action

### PR DESCRIPTION
**Pull request by jeremiahsm-docusign**
Creates a brand new action based on the exist copilot list envelopes action
[Flow-650](https://docusign.atlassian.net/jira/software/c/projects/STRATEGIC/boards/3160?selectedIssue=FLOW-650)

---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [x] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: certified-connectors/DocuSignDemo/script.csx [hunk 1]
```diff
@@ -4156,6 +4156,32 @@ private void RenameSpecificKeys(JObject jObject, Dictionary<string, string> keyM
...
+      var uriBuilder = new UriBuilder(this.Context.Request.RequestUri);
+      var query = HttpUtility.ParseQueryString(this.Context.Request.RequestUri.Query);
+      uriBuilder.Path = uriBuilder.Path.Replace("/SearchListEnvelopes", "");
+
+      query["include"] = "custom_fields, recipients, documents, folders";
+      query["order"] = "desc";
+      
+      query["status"] = string.IsNullOrEmpty(query.Get("envelopeStatus")) ? 
+        null : query.Get("envelopeStatus");
+      query["folder_ids"] = string.IsNullOrEmpty(query.Get("folder_ids")) ?
```
Comment (youngl-docusign) [inline certified-connectors/DocuSignDemo/script.csx:4170]: Is this necessary?

File: certified-connectors/DocuSignDemo/script.csx [hunk 2]
```diff
@@ -5825,6 +5856,81 @@ private void RenameSpecificKeys(JObject jObject, Dictionary<string, string> keyM
...
+            break;
+          }
+        }
+      }
+
+      filteredEnvelopesDetails = GetFilteredEnvelopeDetails(envelopes);
+      newBody["value"] = (filteredEnvelopesDetails.Count < top) ? 
+        filteredEnvelopesDetails : 
+        new JArray(filteredEnvelopesDetails.Skip(skip).Take(top).ToArray());
+      newBody["hasMoreResults"] = (skip + top < filteredEnvelopesDetails.Count) ? true : false;
```
Comment (youngl-docusign) [inline certified-connectors/DocuSignDemo/script.csx:5930]: Help me understand the logic, what is the use of "hasMoreResults"?

File: certified-connectors/DocuSignDemo/script.csx (hunk 1 above)
Comment (release-docusign) [reply certified-connectors/DocuSignDemo/script.csx:4170]: I believe you can optionally filter based on status. @jeremiahs-docusign please confirm.

File: certified-connectors/DocuSignDemo/script.csx (hunk 2 above)
Comment (release-docusign) [reply certified-connectors/DocuSignDemo/script.csx:5930]: So when you get a list of envelopes, you get about 1000 at a time. If there are more than 1000 envelopes in your query, hasMoreResults will say true, and you can follow up with a subsequent call.

**Review by release-docusign — APPROVED** [review]
Please share your test matrix.

File: certified-connectors/DocuSignDemo/script.csx (hunk 1 above)
Comment (jeremiahsm-docusign) [reply certified-connectors/DocuSignDemo/script.csx:4170]: yeah envelope status is one of the filter options.

**Review by release-docusign — DISMISSED** [review]
Will re-approve using non-admin creds.

**Review by angelga-docusign — APPROVED** [review]

File: certified-connectors/DocuSignDemo/script.csx (hunk 1 above)
Comment (youngl-docusign) [reply certified-connectors/DocuSignDemo/script.csx:4170]: If we don't have this line, it `folder_id` would still be included in query as long as user adds that input when using this action correct?

File: certified-connectors/DocuSignDemo/script.csx (hunk 2 above)
Comment (youngl-docusign) [reply certified-connectors/DocuSignDemo/script.csx:5930]: Nice. Got it.

File: certified-connectors/DocuSignDemo/script.csx (hunk 1 above)
Comment (jeremiahsm-docusign) [reply certified-connectors/DocuSignDemo/script.csx:4170]: I removed it and found that when filtering based on status it doesnt pick up on the query for the output. So its necessary to include.

**Comment by jeremiahsm-docusign** [comment]
https://docusign.atlassian.net/wiki/spaces/PAR/pages/594085777/Flow-650+-+Search+List+Envelopes
Test Matrix @angelga-docusign

File: certified-connectors/DocuSignDemo/script.csx (hunk 1 above)
Comment (youngl-docusign) [reply certified-connectors/DocuSignDemo/script.csx:4170]: hmm got it

**Review by youngl-docusign — APPROVED** [review]

**Pull request merged**